### PR TITLE
Optimise ConcurrentQueue

### PR DESCRIPTION
**Pull request by Chaoba**
Optimise ConcurrentQueue and add test cases

### CODE REVIEW

File: libtelemetry/src/main/java/com/mapbox/android/telemetry/ConcurrentQueue.java [hunk 1]
```diff
@@ -8,35 +8,32 @@
 
 class ConcurrentQueue<T> {
```
Comment (andrlee) [inline libtelemetry/src/main/java/com/mapbox/android/telemetry/ConcurrentQueue.java]: IMHO this class should be put to bed completely, i don't see a reason in maintaining additional generic wrapper for `ConcurrentLinkedQueue`

File: libtelemetry/src/main/java/com/mapbox/android/telemetry/FlushQueueCallback.java [hunk 2]
```diff
@@ -3,5 +3,5 @@
 
 interface FlushQueueCallback {
```
Comment (andrlee) [inline libtelemetry/src/main/java/com/mapbox/android/telemetry/FlushQueueCallback.java]: same here, i think this interface is redundant, should be put to bed as well.

Comment (Chaoba) [reply libtelemetry/src/main/java/com/mapbox/android/telemetry/FlushQueueCallback.java]: Yeah, I have the same thought when I read here

**Comment by Chaoba** [comment]
Close since we will refactor this part.

**Pull request closed without merging**